### DISCUSSION TITLE
Is it possible to do Google OAuth in react native with supabase?

### DISCUSSION THREAD

**Discussion by williamlmao** [supabase/supabase#17998] [Questions]
I'm wondering if it is even possible to set up google O Auth with supabase. 

I watched this video: https://www.youtube.com/watch?v=vojHmGUGUGc which shows setup for android, but not ios. 

I also found this issue thread: https://github.com/react-native-google-signin/google-signin/issues/1176

I'm using react-native-google-signin

I've gotten to the point where the sign in flow from google is completed, but saving the user to supabase is where the flow breaks. 

[AuthApiError: Nonces mismatch] {"session": null, "user": null} is the error I get.

**Comment by s0h311** [comment]
Hi @williamlmao, make sure you are calling the `signInWithOAuth` correctly: https://supabase.com/docs/reference/javascript/auth-signinwithoauth. Also you must enable `oAuth` in supabase bashboard under Authentication > Providers.

Edit: Make sure that you have the right URL configured. See Authentication > URL Configuration in supabase dashboard

![grafik](https://github.com/supabase/supabase/assets/113988347/cb130de9-c138-498a-9cab-2925cdf52db0)

**Comment by wishup-agrit** [comment]
I am also facing this error. Is there any resource I can look up to and debug it.

**Reply by ahmed-osman3** [reply]
Did you manage to find a solution?
